datafusion/datasource/src/url.rs

```diff
@@ -628,4 +630,165 @@ mod tests {
             "file path ends with .ext - extension is ext",
         );
     }
+
+    #[tokio::test]
+    async fn test_list_files() {
```

Comment:
I added new tests

I verified these tests fail without the code changes in this PR

```
called `Result::unwrap()` on an `Err` value: ObjectStore(NotFound { path: "t", source: NoDataInMemory { path: "t" } })
thread 'url::tests::test_list_files' panicked at datafusion/datasource/src/url.rs:697:14:
called `Result::unwrap()` on an `Err` value: ObjectStore(NotFound { path: "t", source: NoDataInMemory { path: "t" } })
stack backtrace:
```